domain/src/main/java/com/teamagam/gimelgimel/domain/map/DisplayVectorLayersInteractor.java

```diff
@@ -37,33 +38,55 @@ public DisplayVectorLayersInteractor(
         mVectorLayersVisibilityRepository = vectorLayersVisibilityRepository;
         mLayersLocalCache = layersLocalCache;
         mDisplayer = displayer;
+        mDisplayedVLs = new TreeSet<>();
     }
 
     @Override
     protected SubscriptionRequest buildSubscriptionRequest(
             DisplaySubscriptionRequest.DisplaySubscriptionRequestFactory factory) {
         return factory.create(
-                mVectorLayersVisibilityRepository.getVisibilityChangesLogObservable(),
-                getVectorLayerDisplayAction());
+                mVectorLayersVisibilityRepository.getVisibilityChangesLogObservable()
+                        .map(this::createVectorLayerPresentation),
+                this::display);
+    }
+
+    private VectorLayerPresentation createVectorLayerPresentation(
+            VectorLayerVisibilityChange visibilityChange) {
+        VectorLayer vl = mVectorLayersRepository.get(visibilityChange.getVectorLayerId());
+        URI cachedURI = mLayersLocalCache.getCachedURI(vl);
+        if (visibilityChange.getVisibility()) {
+            return VectorLayerPresentation.createShown(vl, cachedURI);
+        }
+        return VectorLayerPresentation.createHidden(vl, cachedURI);
     }
 
-    private Action1<VectorLayerVisibilityChange> getVectorLayerDisplayAction() {
-        return change -> display(
-                mVectorLayersRepository.get(change.getVectorLayerId()), change.getVisibility());
+    private void display(VectorLayerPresentation vlp) {
+        if (isUpdatingShown(vlp)) {
+            hidePreviousLayer(vlp);
+        }
+        mDisplayer.display(vlp);
+        updateDisplayStatus(vlp);
     }
 
-    private void display(VectorLayer vectorLayer, boolean isVisible) {
-        URI localUri = mLayersLocalCache.getCachedURI(vectorLayer);
-        if (isVisible) {
-            mDisplayer.displayShown(VectorLayerPresentation.create(vectorLayer, localUri));
+    private boolean isUpdatingShown(VectorLayerPresentation vlp) {
```

Comment:
Consider renaming to "isReplacingRequest" or "isShowButAlreadyShown". I'm not sure these are good either but I find "isUpdatingShown" confusing...